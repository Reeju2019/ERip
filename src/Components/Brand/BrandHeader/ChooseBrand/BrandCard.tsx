import React from 'react'
import './ChooseBrand.css'
import { useNavigate } from 'react-router-dom'
interface IBrandCard {
  brand: {
    brandName: string
    image: string
  }
}

const BrandCard: React.FunctionComponent<IBrandCard> = (brand) => {
  const nevigate = useNavigate()
  let otherCheck = false
  if (brand.brand.brandName === 'others') {
    otherCheck = true
  }
  const device: string = localStorage.getItem('deviceId') as string
  const brandClick = () => {
    localStorage.setItem('brandId', JSON.stringify(brand.brand.brandName))

    const deviceId = JSON.parse(device)
    // console.log(deviceId)

    nevigate(`/${deviceId}/${brand.brand.brandName}`)
  }

  return (
    <>
      <div onClick={brandClick} className={'apple brand-width ' + (otherCheck ? 'other-img' : '')}>
        <img src={brand.brand.image} alt={brand.brand.brandName} className='brand-img' />
      </div>
    </>
  )
}

export default BrandCard
