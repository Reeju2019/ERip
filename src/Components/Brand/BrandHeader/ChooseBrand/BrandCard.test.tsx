import { render } from '@testing-library/react'
import BrandCard from './BrandCard'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'

interface Brand {
  brandName: string
  image: string
}
const brand: Brand = {
  brandName: 'apple',
  image: '',
}

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}))

describe('App rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<BrandCard brand={brand} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(<BrandCard brand={brand} />)
    ReactDOM.unmountComponentAtNode(div)
  })
})
