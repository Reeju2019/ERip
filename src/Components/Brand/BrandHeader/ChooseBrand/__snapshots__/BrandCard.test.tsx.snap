// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering specification SnapShot testing 1`] = `
<div
  className="apple brand-width "
  onClick={[Function]}
>
  <img
    alt="apple"
    className="brand-img"
    src=""
  />
</div>
`;
